metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/MetadataChangeLogProcessor.java

```diff
@@ -51,18 +49,60 @@ public MetadataChangeLogProcessor(
       @Nonnull final EntityChangeEventGeneratorHook entityChangeEventHook,
       @Nonnull final SiblingAssociationHook siblingAssociationHook
   ) {
-    this.hooks = ImmutableList.of(updateIndicesHook, ingestionSchedulerHook, entityChangeEventHook, siblingAssociationHook);
-    this.hooks.forEach(MetadataChangeLogHook::init);
+    hookMap = new HashMap<>();
+    // Here - plug in additional "custom processor hooks" and create handler
+    hookMap.put(updateIndicesHook.getClass(), updateIndicesHook);
+    hookMap.put(ingestionSchedulerHook.getClass(), ingestionSchedulerHook);
+    hookMap.put(entityChangeEventHook.getClass(), entityChangeEventHook);
+    hookMap.put(siblingAssociationHook.getClass(), siblingAssociationHook);
+  }
+
+  @KafkaListener(id = "${METADATA_CHANGE_LOG_KAFKA_UPDATE_INDICES_CONSUMER_GROUP_ID:"
+          + "generic-mae-consumer-job-client-update-indices-hook}", topics = {
```

Comment:
I see the benefits in allowing separate offsets for each hook independently which allows for per hook recovery, however I am concerned that this will create out of order issues when the hooks need to be applied in sync. I would suggest considering a way to group related hooks together when needed within the same consumer group to maintain backwards compatibility. If the hooks can be demonstrated to be independent, then move them to separate consumer groups.

Reply:
@djordje-mijatovic what is your thought process regarding separate consumer groups per consume?

Reply:
Currently there are 4 different hooks. If one hook throws an exception, this handle will be skipped forever. There is no way to retry only one hook - the one that failed.
With for different Kafka listeners (one per hook) we are able to retry only the failed hook without executing correct hooks again.

Reply:
>> I am concerned that this will create out of order issues when the hooks need to be applied in sync.
Do we have this case? Even in this situation, what do you think about having a single hook doing two different things, instead of having a hooks in a group?

Reply:
@david-leifker 
I was thinking about how to implement it. The only way I see it, is to still have defined 4 different listeners (4 different consumer groups) but allowing hook configuration for all of them.

Instead of having consumer groups like: 

```
generic-mae-consumer-job-client-update-indices-hook
generic-mae-consumer-job-client-ingestion-scheduler-hook
generic-mae-consumer-job-client-entity-change-event-generator-hook
generic-mae-consumer-job-client-sibling-association-hook
```

we should rename them into some more generic names like:
```
generic-mae-consumer-job-client-hook_group_1
generic-mae-consumer-job-client-hook_group_2
generic-mae-consumer-job-client-hook_group_3
generic-mae-consumer-job-client-hook_group_4
```

For every consumer group we should have a parameter in yml file defining which hooks should be called and we which order.

If we want to support the original implementation (all 4 hooks in one listener) we should have settings like this:
```
kafka.mae.consumer.group.1=updateIndicesHook,ingestionSchedulerHook,entityChangeEventHook,siblingAssociationHook
kafka.mae.consumer.group.2=
kafka.mae.consumer.group.3=
kafka.mae.consumer.group.4=
```

In case we want have all 4 hooks in different groups we should have settings like this:
```
kafka.mae.consumer.group.1=updateIndicesHook
kafka.mae.consumer.group.2=ingestionSchedulerHook
kafka.mae.consumer.group.3=entityChangeEventHook
kafka.mae.consumer.group.4=siblingAssociationHook
```

At the if we want to have updateIndicesHook and ingestionSchedulerHook in the same listener those are expected settings:
```
kafka.mae.consumer.group.1=updateIndicesHook, ingestionSchedulerHook
kafka.mae.consumer.group.2=entityChangeEventHook
kafka.mae.consumer.group.3=siblingAssociationHook
kafka.mae.consumer.group.4=
```